core/src/main/java/org/elasticsearch/ElasticsearchRemoteException.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Exception thrown when a remote action fails. Allows settings an arbitrary {@link RestStatus} so the remote failures that include status
+ * information can be returned without losing anything.
+ */
+public class ElasticsearchRemoteException extends ElasticsearchException {
+    private final RestStatus status;
+
+    /**
+     * Build the exception with a specific status and cause.
+     */
+    public ElasticsearchRemoteException(String msg, RestStatus status, Throwable cause, Object... args) {
+        super(msg, cause, args);
+        this.status = status;
+    }
+
+    /**
+     * Build the exception with the status derrived from the cause.
+     */
+    public ElasticsearchRemoteException(String msg, Exception cause, Object... args) {
+        this(msg, ExceptionsHelper.status(cause), cause, args);
+    }
+
+    /**
+     * Build the exception without a cause.
+     */
+    public ElasticsearchRemoteException(String msg, RestStatus status, Object... args) {
```

Comment:
this seems to be used only in a test. can we drop it and make up some cause in that test?